### ISSUE TITLE
After switching ngx-default from sql to sqlfull default app is not passing Sonar quality gate

### ISSUE THREAD

**Issue by kaidohallik** [jhipster/generator-jhipster#13291]
##### **Overview of the issue**

After adding tests to entity routing resolve service I hoped to get better coverage, but instead yesterday evening change in https://sonarcloud.io/dashboard?id=jhipster-sample-application:
Coverage: 81.0 -> 75.8
Files: 593 -> 1,173
Unit tests: 762 -> 1,192
Lines to Cover: 10,283 -> 18,072
Code smells: 74 -> 156

If I analyze the same project in local SonarQube then results:
Coverage: 82.5
Files: 595
Unit tests: 762
Lines to Cover: 10,457
Code smells: 74

If I look at code smell details in https://sonarcloud.io/ then can see that java test files from `src/test/` are analyzed (although `sonar.sources=src/main/`).
Maybe a new version of SonarCloud was deployed yesterday evening which interprets properties from `sonar-project.properties` differently.

##### **Motivation for or Use Case**

Test files should not be analyzed.

##### **Reproduce the error**

Not reproducible in SonarQube, only SonarCloud is affected.

##### **Related issues**

#13232

##### **Suggest a Fix**

I put back `sonar.tests=src/test/` also for projects with Angular client with PR coming shortly, maybe this helps in SonarCloud.

##### **JHipster Version(s)**

```
jhipster-sample-application@0.0.1-SNAPSHOT c:\projects\teest\jhipster-sonar
`-- generator-jhipster@6.10.5  -> c:\projects\kaido-github\generator-jhipster

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "jhipsterSampleApplication",
    "packageName": "tech.jhipster.sample",
    "packageFolder": "tech/jhipster/sample",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "buildTool": "maven",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "al",
      "ar-ly",
      "hy",
      "by",
      "bn",
      "bg",
      "ca",
      "zh-cn",
      "zh-tw",
      "cs",
      "da",
      "nl",
      "et",
      "fa",
      "fr",
      "gl",
      "de",
      "el",
      "hi",
      "hr",
      "hu",
      "in",
      "it",
      "ja",
      "ko",
      "mr",
      "my",
      "pl",
      "pt-br",
      "pt-pt",
      "ro",
      "ru",
      "si",
      "sk",
      "sr",
      "es",
      "sv",
      "tr",
      "ta",
      "te",
      "th",
      "ua",
      "uz-Cyrl-uz",
      "uz-Latn-uz",
      "vi"
    ],
    "testFrameworks": ["gatling", "cypress"],
    "serverPort": "8080",
    "jhiPrefix": "myPrefix",
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "entitySuffix": "Entity",
    "dtoSuffix": "",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "embeddableLaunchScript": false,
    "clientTheme": "none",
    "otherModules": [],
    "blueprints": [],
    "creationTimestamp": 1596513172471,
    "skipUserManagement": false,
    "withAdminUi": true,
    "clientThemeVariant": "",
    "skipClient": false,
    "skipServer": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "pages": [],
    "reactive": false,
    "jhipsterVersion": "6.10.5",
    "entities": [
      "Operation",
      "FieldTestEntity",
      "FieldTestInfiniteScrollEntity",
      "FieldTestMapstructAndServiceClassEntity",
      "FieldTestPaginationEntity",
      "FieldTestServiceClassAndJpaFilteringEntity",
      "FieldTestServiceImplEntity",
      "EntityWithDTO",
      "EntityWithServiceClassAndPagination",
      "EntityWithServiceImplAndPagination",
      "EntityWithServiceImplAndDTO",
      "EntityWithPaginationAndDTO",
      "EntityWithServiceClassPaginationAndDTO",
      "EntityWithServiceImplPaginationAndDTO",
      "MapsIdUserProfileWithDTO",
      "FieldTestEnumWithValue",
      "BankAccount",
      "Label"
    ],
    "lastLiquibaseTimestamp": 1607595190000
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Operation {
  date Instant required
  description String
  amount BigDecimal required
}
entity FieldTestEntity {
  stringTom String
  stringRequiredTom String required
  stringMinlengthTom String minlength(0)
  stringMaxlengthTom String maxlength(20)
  stringPatternTom String pattern(/^[a-zA-Z0-9]*$/)
  numberPatternTom String pattern(/^[0-9]+$/)
  numberPatternRequiredTom String pattern(/^[0-9]+$/) required
  integerTom Integer
  integerRequiredTom Integer required
  integerMinTom Integer min(0)
  integerMaxTom Integer max(100)
  longTom Long
  longRequiredTom Long required
  longMinTom Long min(0)
  longMaxTom Long max(100)
  floatTom Float
  floatRequiredTom Float required
  floatMinTom Float min(0)
  floatMaxTom Float max(100)
  doubleRequiredTom Double required
  doubleMinTom Double min(0)
  doubleMaxTom Double max(100)
  bigDecimalRequiredTom BigDecimal required
  bigDecimalMinTom BigDecimal min(0)
  bigDecimalMaxTom BigDecimal max(100)
  localDateTom LocalDate
  localDateRequiredTom LocalDate required
  instantTom Instant
  instantRequiredTom Instant required
  zonedDateTimeTom ZonedDateTime
  zonedDateTimeRequiredTom ZonedDateTime required
  durationTom Duration
  durationRequiredTom Duration required
  booleanTom Boolean
  booleanRequiredTom Boolean required
  enumTom EnumFieldClass
  enumRequiredTom EnumRequiredFieldClass required
  uuidTom UUID
  uuidRequiredTom UUID required
  byteImageTom ImageBlob
  byteImageRequiredTom ImageBlob required
  byteImageMinbytesTom ImageBlob minbytes(0)
  byteImageMaxbytesTom ImageBlob maxbytes(10000)
  byteAnyTom AnyBlob
  byteAnyRequiredTom AnyBlob required
  byteAnyMinbytesTom AnyBlob minbytes(0)
  byteAnyMaxbytesTom AnyBlob maxbytes(10000)
  byteTextTom TextBlob
  byteTextRequiredTom TextBlob required
}
entity FieldTestInfiniteScrollEntity {
  stringHugo String
  stringRequiredHugo String required
  stringMinlengthHugo String minlength(0)
  stringMaxlengthHugo String maxlength(20)
  stringPatternHugo String pattern(/^[a-zA-Z0-9]*$/)
  integerHugo Integer
  integerRequiredHugo Integer required
  integerMinHugo Integer min(0)
  integerMaxHugo Integer max(100)
  longHugo Long
  longRequiredHugo Long required
  longMinHugo Long min(0)
  longMaxHugo Long max(100)
  floatHugo Float
  floatRequiredHugo Float required
  floatMinHugo Float min(0)
  floatMaxHugo Float max(100)
  doubleRequiredHugo Double required
  doubleMinHugo Double min(0)
  doubleMaxHugo Double max(100)
  bigDecimalRequiredHugo BigDecimal required
  bigDecimalMinHugo BigDecimal min(0)
  bigDecimalMaxHugo BigDecimal max(100)
  localDateHugo LocalDate
  localDateRequiredHugo LocalDate required
  instantHugo Instant
  instanteRequiredHugo Instant required
  zonedDateTimeHugo ZonedDateTime
  zonedDateTimeRequiredHugo ZonedDateTime required
  durationHugo Duration
  durationRequiredHugo Duration required
  booleanHugo Boolean
  booleanRequiredHugo Boolean required
  enumHugo EnumFieldClass
  enumRequiredHugo EnumRequiredFieldClass required
  uuidHugo UUID
  uuidRequiredHugo UUID required
  byteImageHugo ImageBlob
  byteImageRequiredHugo ImageBlob required
  byteImageMinbytesHugo ImageBlob minbytes(0)
  byteImageMaxbytesHugo ImageBlob maxbytes(10000)
  byteAnyHugo AnyBlob
  byteAnyRequiredHugo AnyBlob required
  byteAnyMinbytesHugo AnyBlob minbytes(0)
  byteAnyMaxbytesHugo AnyBlob maxbytes(10000)
  byteTextHugo TextBlob
  byteTextRequiredHugo TextBlob required
}
entity FieldTestMapstructAndServiceClassEntity {
  stringEva String
  stringRequiredEva String required
  stringMinlengthEva String minlength(0)
  stringMaxlengthEva String maxlength(20)
  stringPatternEva String pattern(/^[a-zA-Z0-9]*$/)
  integerEva Integer
  integerRequiredEva Integer required
  integerMinEva Integer min(0)
  integerMaxEva Integer max(100)
  longEva Long
  longRequiredEva Long required
  longMinEva Long min(0)
  longMaxEva Long max(100)
  floatEva Float
  floatRequiredEva Float required
  floatMinEva Float min(0)
  floatMaxEva Float max(100)
  doubleRequiredEva Double required
  doubleMinEva Double min(0)
  doubleMaxEva Double max(100)
  bigDecimalRequiredEva BigDecimal required
  bigDecimalMinEva BigDecimal min(0)
  bigDecimalMaxEva BigDecimal max(100)
  localDateEva LocalDate
  localDateRequiredEva LocalDate required
  instantEva Instant
  instanteRequiredEva Instant required
  zonedDateTimeEva ZonedDateTime
  zonedDateTimeRequiredEva ZonedDateTime required
  durationEva Duration
  durationRequiredEva Duration required
  booleanEva Boolean
  booleanRequiredEva Boolean required
  enumEva EnumFieldClass
  enumRequiredEva EnumRequiredFieldClass required
  uuidEva UUID
  uuidRequiredEva UUID required
  byteImageEva ImageBlob
  byteImageRequiredEva ImageBlob required
  byteImageMinbytesEva ImageBlob minbytes(0)
  byteImageMaxbytesEva ImageBlob maxbytes(10000)
  byteAnyEva AnyBlob
  byteAnyRequiredEva AnyBlob required
  byteAnyMinbytesEva AnyBlob minbytes(0)
  byteAnyMaxbytesEva AnyBlob maxbytes(10000)
  byteTextEva TextBlob
  byteTextRequiredEva TextBlob required
}
entity FieldTestPaginationEntity {
  stringAlice String
  stringRequiredAlice String required
  stringMinlengthAlice String minlength(0)
  stringMaxlengthAlice String maxlength(20)
  stringPatternAlice String pattern(/^[a-zA-Z0-9]*$/)
  integerAlice Integer
  integerRequiredAlice Integer required
  integerMinAlice Integer min(0)
  integerMaxAlice Integer max(100)
  longAlice Long
  longRequiredAlice Long required
  longMinAlice Long min(0)
  longMaxAlice Long max(100)
  floatAlice Float
  floatRequiredAlice Float required
  floatMinAlice Float min(0)
  floatMaxAlice Float max(100)
  doubleRequiredAlice Double required
  doubleMinAlice Double min(0)
  doubleMaxAlice Double max(100)
  bigDecimalRequiredAlice BigDecimal required
  bigDecimalMinAlice BigDecimal min(0)
  bigDecimalMaxAlice BigDecimal max(100)
  localDateAlice LocalDate
  localDateRequiredAlice LocalDate required
  instantAlice Instant
  instanteRequiredAlice Instant required
  zonedDateTimeAlice ZonedDateTime
  zonedDateTimeRequiredAlice ZonedDateTime required
  durationAlice Duration
  durationRequiredAlice Duration required
  booleanAlice Boolean
  booleanRequiredAlice Boolean required
  enumAlice EnumFieldClass
  enumRequiredAlice EnumRequiredFieldClass required
  uuidAlice UUID
  uuidRequiredAlice UUID required
  byteImageAlice ImageBlob
  byteImageRequiredAlice ImageBlob required
  byteImageMinbytesAlice ImageBlob minbytes(0)
  byteImageMaxbytesAlice ImageBlob maxbytes(10000)
  byteAnyAlice AnyBlob
  byteAnyRequiredAlice AnyBlob required
  byteAnyMinbytesAlice AnyBlob minbytes(0)
  byteAnyMaxbytesAlice AnyBlob maxbytes(10000)
  byteTextAlice TextBlob
  byteTextRequiredAlice TextBlob required
}
entity FieldTestServiceClassAndJpaFilteringEntity {
  stringBob String
  stringRequiredBob String required
  stringMinlengthBob String minlength(0)
  stringMaxlengthBob String maxlength(20)
  stringPatternBob String pattern(/^[a-zA-Z0-9]*$/)
  integerBob Integer
  integerRequiredBob Integer required
  integerMinBob Integer min(0)
  integerMaxBob Integer max(100)
  longBob Long
  longRequiredBob Long required
  longMinBob Long min(0)
  longMaxBob Long max(100)
  floatBob Float
  floatRequiredBob Float required
  floatMinBob Float min(0)
  floatMaxBob Float max(100)
  doubleRequiredBob Double required
  doubleMinBob Double min(0)
  doubleMaxBob Double max(100)
  bigDecimalRequiredBob BigDecimal required
  bigDecimalMinBob BigDecimal min(0)
  bigDecimalMaxBob BigDecimal max(100)
  localDateBob LocalDate
  localDateRequiredBob LocalDate required
  instantBob Instant
  instanteRequiredBob Instant required
  zonedDateTimeBob ZonedDateTime
  zonedDateTimeRequiredBob ZonedDateTime required
  durationBob Duration
  durationRequiredBob Duration required
  booleanBob Boolean
  booleanRequiredBob Boolean required
  enumBob EnumFieldClass
  enumRequiredBob EnumRequiredFieldClass required
  uuidBob UUID
  uuidRequiredBob UUID required
  byteImageBob ImageBlob
  byteImageRequiredBob ImageBlob required
  byteImageMinbytesBob ImageBlob minbytes(0)
  byteImageMaxbytesBob ImageBlob maxbytes(10000)
  byteAnyBob AnyBlob
  byteAnyRequiredBob AnyBlob required
  byteAnyMinbytesBob AnyBlob minbytes(0)
  byteAnyMaxbytesBob AnyBlob maxbytes(10000)
  byteTextBob TextBlob
  byteTextRequiredBob TextBlob required
}
entity FieldTestServiceImplEntity {
  stringMika String
  stringRequiredMika String required
  stringMinlengthMika String minlength(0)
  stringMaxlengthMika String maxlength(20)
  stringPatternMika String pattern(/^[a-zA-Z0-9]*$/)
  integerMika Integer
  integerRequiredMika Integer required
  integerMinMika Integer min(0)
  integerMaxMika Integer max(100)
  longMika Long
  longRequiredMika Long required
  longMinMika Long min(0)
  longMaxMika Long max(100)
  floatMika Float
  floatRequiredMika Float required
  floatMinMika Float min(0)
  floatMaxMika Float max(100)
  doubleRequiredMika Double required
  doubleMinMika Double min(0)
  doubleMaxMika Double max(100)
  bigDecimalRequiredMika BigDecimal required
  bigDecimalMinMika BigDecimal min(0)
  bigDecimalMaxMika BigDecimal max(100)
  localDateMika LocalDate
  localDateRequiredMika LocalDate required
  instantMika Instant
  instanteRequiredMika Instant required
  zonedDateTimeMika ZonedDateTime
  zonedDateTimeRequiredMika ZonedDateTime required
  durationMika Duration
  durationRequiredMika Duration required
  booleanMika Boolean
  booleanRequiredMika Boolean required
  enumMika EnumFieldClass
  enumRequiredMika EnumRequiredFieldClass required
  uuidMika UUID
  uuidRequiredMika UUID required
  byteImageMika ImageBlob
  byteImageRequiredMika ImageBlob required
  byteImageMinbytesMika ImageBlob minbytes(0)
  byteImageMaxbytesMika ImageBlob maxbytes(10000)
  byteAnyMika AnyBlob
  byteAnyRequiredMika AnyBlob required
  byteAnyMinbytesMika AnyBlob minbytes(0)
  byteAnyMaxbytesMika AnyBlob maxbytes(10000)
  byteTextMika TextBlob
  byteTextRequiredMika TextBlob required
}
entity EntityWithDTO {
  emma String
}
entity EntityWithServiceClassAndPagination {
  enzo String
}
entity EntityWithServiceImplAndPagination {
  hugo String
}
entity EntityWithServiceImplAndDTO {
  louis String
}
entity EntityWithPaginationAndDTO {
  lea String
}
entity EntityWithServiceClassPaginationAndDTO {
  lena String
}
entity EntityWithServiceImplPaginationAndDTO {
  theo String
}
entity MapsIdUserProfileWithDTO (maps_id_user_profile_withdto) {
  dateOfBirth Instant
}
entity FieldTestEnumWithValue (entity_with_enums) {
  myFieldA MyEnumA
  myFieldB MyEnumB
  myFieldC MyEnumC
}
entity BankAccount {
  name String required
  guid UUID
  bankNumber Integer
  agencyNumber Long
  lastOperationDuration Float
  meanOperationDuration Double
  meanQueueDuration Duration
  balance BigDecimal required
  openingDay LocalDate
  lastOperationDate Instant
  active Boolean
  accountType BankAccountType
  attachment AnyBlob
  description TextBlob
}
entity Label {
  labelName String required minlength(3)
}
enum EnumFieldClass {
  ENUM_VALUE_1,
  ENUM_VALUE_2,
  ENUM_VALUE_3
}
enum EnumRequiredFieldClass {
  ENUM_VALUE_1,
  ENUM_VALUE_2,
  ENUM_VALUE_3
}
enum MyEnumA {
  AAA,
  BBB
}
enum MyEnumB {
  AAA (aaa_aaa),
  BBB
}
enum MyEnumC {
  AAA (aaa_aaa),
  BBB (b)
}
enum BankAccountType {
  CHECKING,
  SAVINGS,
  LOAN
}

relationship OneToOne {
  MapsIdUserProfileWithDTO{user(login)} to User
}
relationship OneToMany {
  BankAccount{operation} to Operation{bankAccount(name)}
}
relationship ManyToOne {
  BankAccount{user(login)} to User
}
relationship ManyToMany {
  Operation{label(labelName)} to Label{operation}
}

paginate Operation, FieldTestInfiniteScrollEntity with infinite-scroll
paginate FieldTestPaginationEntity, EntityWithServiceClassAndPagination, EntityWithServiceImplAndPagination, EntityWithPaginationAndDTO, EntityWithServiceClassPaginationAndDTO, EntityWithServiceImplPaginationAndDTO, Label with pagination
clientRootFolder Operation, BankAccount, Label with test-root
dto FieldTestMapstructAndServiceClassEntity, EntityWithDTO, EntityWithServiceImplAndDTO, EntityWithPaginationAndDTO, EntityWithServiceClassPaginationAndDTO, EntityWithServiceImplPaginationAndDTO, MapsIdUserProfileWithDTO with mapstruct
service FieldTestMapstructAndServiceClassEntity, FieldTestServiceClassAndJpaFilteringEntity, EntityWithDTO, EntityWithServiceClassAndPagination, EntityWithPaginationAndDTO, EntityWithServiceClassPaginationAndDTO, Label with serviceClass
service FieldTestServiceImplEntity, EntityWithServiceImplAndPagination, EntityWithServiceImplAndDTO, EntityWithServiceImplPaginationAndDTO, MapsIdUserProfileWithDTO, BankAccount with serviceImpl
filter FieldTestServiceClassAndJpaFilteringEntity, BankAccount
angularSuffix BankAccount with mySuffix
readOnly Label

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)

git version 2.29.2.windows.3

node: v14.15.1

npm: 6.14.9

yeoman: 3.1.1

Docker version 20.10.0, build 7287ab3

docker-compose version 1.27.4, build 40524192

##### **Browsers and Operating System**

Windows 10

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by kaidohallik** [comment]
SonarQube 8.6.0 is out and 12 hours ago Docker image is available in Docker Hub. Will try this new version locally.

**Comment by DanielFran** [comment]
Will update current docker images, including Sonar asap...

**Comment by kaidohallik** [comment]
SonarQube 8.6.0 needs authentication by default - so this needs to be handled while upgrading SonarQube docker image to 8.6.0.

**Comment by kaidohallik** [comment]
This is needed to add to `sonar-project.properties` for SonarQube 8.6.0:
```
# After changing password in SonarQube change this also here
sonar.login=admin
sonar.password=admin
```
In SonarQube 8.6.0 all is working well, the same result as in SonarQube 8.5.1

**Comment by DanielFran** [comment]
@kaidohallik the changes in sonar-project.properties are needed if we use a token?

Or you mean, just add this to document sonar-project.properties generated?

**Comment by kaidohallik** [comment]
This is needed for local SonarQube, if not providing then the following commands will fail:

Maven:
```
./mvnw -Pprod clean verify sonar:sonar
```

Gradle:
```
./gradlew -Pprod clean check jacocoTestReport sonarqube
```

SonarCloud is working without this. If we add this then don't know, will SonarCloud integration work or not. I believe that for SonarCloud token is priority, so SonarCloud probably should also work after adding this.

**Comment by kaidohallik** [comment]
This issue is related to: https://github.com/jhipster/generator-jhipster/commit/a65bba5a212609fe25fdc5e3d0b1df3d32303bd7 - sample application was switched from `sql` to `sqlfull` by this change - so more entities and this affected also coverage.

**Comment by kaidohallik** [comment]
As detecting code smells of test files is normal part of Sonar then this issue is only for coverage which appeared after switching ngx-default from `sql` to `sqlfull`, changed also issue title.

**Comment by mshima** [comment]
That was an unexpected result.
We switched to postgres by default, made sense to switch sqlfull to postgres database.
Curious if there is a way to add sonar status to PRs for our use case.

**Comment by pascalgrimaud** [comment]
still needs to improve our Sonar analysis

**Comment by kaidohallik** [comment]
https://sonarcloud.io/dashboard?id=jhipster-sample-application passes now Sonar quality gate. It's probably after #13592. So closing this issue as resolved.
If someone wants to work with idea to add Sonar check to PR-s - look at https://github.com/jhipster/generator-jhipster/issues/13291#issuecomment-745291671 - then I think can open a new issue.
